### ISSUE TITLE
docker-engine-1.10.1-1.el7.centos.x86_64 failed to install on Redhat7.1

### ISSUE THREAD

**Issue by immarvin** [docker/docker#20484]
Description of problem:
I tried to upgrade docker-engine from 1.9.1 to 1.10.1 on Redhat 7.1, the upgrade failed with messages:
````
--> Finished Dependency Resolution
Error: Package: docker-engine-1.10.1-1.el7.centos.x86_64 (dockerrepo)
           Requires: libsystemd.so.0(LIBSYSTEMD_209)(64bit)
Error: Package: docker-engine-1.10.1-1.el7.centos.x86_64 (dockerrepo)
           Requires: libsystemd.so.0()(64bit)
Error: Package: docker-engine-1.10.1-1.el7.centos.x86_64 (dockerrepo)
           Requires: libdevmapper.so.1.02(DM_1_02_97)(64bit)
 You could try using --skip-broken to work around the problem
 You could try running: rpm -Va --nofiles --nodigest
````

`uname -a`:

[root@c910f05c01bc06 tmp]# uname -a
Linux c910f05c01bc06 3.10.0-229.el7.x86_64 #1 SMP Thu Jan 29 18:37:38 EST 2015 x86_64 x86_64 x86_64 GNU/Linux
[root@c910f05c01bc06 tmp]# 

Environment details (AWS, VirtualBox, physical, etc.):
Physical

How reproducible:

Steps to Reproduce:
1. yum clean metadata
2.yum upgrade

Actual Results:
````

[root@c910f05c01bc06 tmp]# yum upgrade       
Loaded plugins: product-id, subscription-manager
This system is not registered to Red Hat Subscription Management. You can use subscription-manager to register.
Resolving Dependencies
--> Running transaction check
---> Package docker-engine.x86_64 0:1.9.1-1.el7.centos will be updated
---> Package docker-engine.x86_64 0:1.10.1-1.el7.centos will be an update
--> Processing Dependency: libdevmapper.so.1.02(DM_1_02_97)(64bit) for package: docker-engine-1.10.1-1.el7.centos.x86_64
--> Processing Dependency: libsystemd.so.0(LIBSYSTEMD_209)(64bit) for package: docker-engine-1.10.1-1.el7.centos.x86_64
--> Processing Dependency: libsystemd.so.0()(64bit) for package: docker-engine-1.10.1-1.el7.centos.x86_64
---> Package docker-engine-selinux.noarch 0:1.9.1-1.el7.centos will be updated
---> Package docker-engine-selinux.noarch 0:1.10.1-1.el7.centos will be an update
---> Package perl-Crypt-Rijndael.x86_64 0:1.09-2.el7 will be updated
---> Package perl-Crypt-Rijndael.x86_64 0:1.12-1.el7 will be an update
---> Package perl-Expect.noarch 0:1.21-1 will be updated
---> Package perl-Expect.noarch 0:1.21-14.el7 will be an update
---> Package perl-SOAP-Lite.noarch 0:0.710.08-1 will be updated
--> Processing Dependency: perl(IO::SessionSet) for package: perl-SOAP-Lite-1.10-1.el7.noarch
--> Processing Dependency: perl(IO::SessionData) for package: perl-SOAP-Lite-1.10-1.el7.noarch
---> Package perl-SOAP-Lite.noarch 0:1.10-1.el7 will be an update
--> Processing Dependency: perl(MIME::Lite) for package: perl-SOAP-Lite-1.10-1.el7.noarch
--> Processing Dependency: perl(MIME::Entity) for package: perl-SOAP-Lite-1.10-1.el7.noarch
--> Processing Dependency: perl(Class::Inspector) for package: perl-SOAP-Lite-1.10-1.el7.noarch
--> Running transaction check
---> Package docker-engine.x86_64 0:1.10.1-1.el7.centos will be an update
--> Processing Dependency: libdevmapper.so.1.02(DM_1_02_97)(64bit) for package: docker-engine-1.10.1-1.el7.centos.x86_64
--> Processing Dependency: libsystemd.so.0(LIBSYSTEMD_209)(64bit) for package: docker-engine-1.10.1-1.el7.centos.x86_64
--> Processing Dependency: libsystemd.so.0()(64bit) for package: docker-engine-1.10.1-1.el7.centos.x86_64
---> Package perl-IO-SessionData.noarch 0:1.03-1.el7 will be installed
---> Package perl-MIME-Lite.noarch 0:3.030-1.el7 will be installed
--> Processing Dependency: perl(MIME::Types) >= 1.28 for package: perl-MIME-Lite-3.030-1.el7.noarch
--> Processing Dependency: perl(Email::Date::Format) for package: perl-MIME-Lite-3.030-1.el7.noarch
---> Package perl-MIME-tools.noarch 0:5.505-1.el7 will be installed
--> Processing Dependency: perl(Mail::Internet) >= 1.28 for package: perl-MIME-tools-5.505-1.el7.noarch
--> Processing Dependency: perl(Mail::Header) >= 1.09 for package: perl-MIME-tools-5.505-1.el7.noarch
--> Processing Dependency: perl(Mail::Field) >= 1.05 for package: perl-MIME-tools-5.505-1.el7.noarch
--> Processing Dependency: perl(Mail::Send) for package: perl-MIME-tools-5.505-1.el7.noarch
--> Processing Dependency: perl(Mail::Field) for package: perl-MIME-tools-5.505-1.el7.noarch
--> Processing Dependency: perl(Convert::BinHex) for package: perl-MIME-tools-5.505-1.el7.noarch
---> Package perl-SOAP-Lite.noarch 0:1.10-1.el7 will be an update
--> Processing Dependency: perl(Class::Inspector) for package: perl-SOAP-Lite-1.10-1.el7.noarch
--> Running transaction check
---> Package docker-engine.x86_64 0:1.10.1-1.el7.centos will be an update
--> Processing Dependency: libdevmapper.so.1.02(DM_1_02_97)(64bit) for package: docker-engine-1.10.1-1.el7.centos.x86_64
--> Processing Dependency: libsystemd.so.0(LIBSYSTEMD_209)(64bit) for package: docker-engine-1.10.1-1.el7.centos.x86_64
--> Processing Dependency: libsystemd.so.0()(64bit) for package: docker-engine-1.10.1-1.el7.centos.x86_64
---> Package perl-Convert-BinHex.noarch 0:1.119-20.el7 will be installed
---> Package perl-Email-Date-Format.noarch 0:1.002-15.el7 will be installed
---> Package perl-MIME-Types.noarch 0:1.38-2.el7 will be installed
---> Package perl-MailTools.noarch 0:2.12-2.el7 will be installed
--> Processing Dependency: perl(Net::SMTP::SSL) for package: perl-MailTools-2.12-2.el7.noarch
---> Package perl-SOAP-Lite.noarch 0:1.10-1.el7 will be an update
--> Processing Dependency: perl(Class::Inspector) for package: perl-SOAP-Lite-1.10-1.el7.noarch
--> Running transaction check
---> Package docker-engine.x86_64 0:1.10.1-1.el7.centos will be an update
--> Processing Dependency: libdevmapper.so.1.02(DM_1_02_97)(64bit) for package: docker-engine-1.10.1-1.el7.centos.x86_64
--> Processing Dependency: libsystemd.so.0(LIBSYSTEMD_209)(64bit) for package: docker-engine-1.10.1-1.el7.centos.x86_64
--> Processing Dependency: libsystemd.so.0()(64bit) for package: docker-engine-1.10.1-1.el7.centos.x86_64
---> Package perl-Net-SMTP-SSL.noarch 0:1.01-13.el7 will be installed
---> Package perl-SOAP-Lite.noarch 0:1.10-1.el7 will be an update
--> Processing Dependency: perl(Class::Inspector) for package: perl-SOAP-Lite-1.10-1.el7.noarch
--> Finished Dependency Resolution
Error: Package: docker-engine-1.10.1-1.el7.centos.x86_64 (dockerrepo)
           Requires: libsystemd.so.0(LIBSYSTEMD_209)(64bit)
Error: Package: docker-engine-1.10.1-1.el7.centos.x86_64 (dockerrepo)
           Requires: libsystemd.so.0()(64bit)
Error: Package: docker-engine-1.10.1-1.el7.centos.x86_64 (dockerrepo)
           Requires: libdevmapper.so.1.02(DM_1_02_97)(64bit)
Error: Package: perl-SOAP-Lite-1.10-1.el7.noarch (epel)
           Requires: perl(Class::Inspector)
 You could try using --skip-broken to work around the problem
 You could try running: rpm -Va --nofiles --nodigest
[root@c910f05c01bc06 tmp]# docker version
````

Expected Results:
upgrade successfully, docker upgraded to 1.10.1

Additional info:
NA

**Comment by HackToday** [comment]
Did you docker repo updated on your redhat system ?

**Comment by immarvin** [comment]
my docker repo:
````
[root@c910f05c01bc06 ~]# cat /etc/yum.repos.d/docker.repo 
[dockerrepo]
name=Docker Repository
baseurl=https://yum.dockerproject.org/repo/main/centos/7
enabled=1
gpgcheck=1
gpgkey=https://yum.dockerproject.org/gpg
````

**Comment by chrishawkins** [comment]
@immarvin Did you find a way around this? I have the same issue but don't have the time to look at it in detail right now.

**Comment by immarvin** [comment]
@chrishawkins , no

**Comment by immarvin** [comment]
hi @chrishawkins, I added the repo of redhat 7.1-updates to yum repolist, then run ``yum upgrade`` , the docker 1.10.2 was installed successfully.

Just a reminder, if you have other docker engine release installed before, after ``yum upgrade, you need to uninstall the previous docker release manually.

**Comment by thaJeztah** [comment]
Isn't that repo enabled by default? I don't have access to a RHEL server, but spinning up a CentOS 7.1 droplet on DigitalOcean gave me this;

```
yum repolist
Loaded plugins: fastestmirror
Loading mirror speeds from cached hostfile
 * base: centos.mirror.transip.nl
 * extras: centos.mirror.transip.nl
 * updates: mirror.amsiohosting.net
```

**Comment by chrishawkins** [comment]
Thanks @immarvin and @thaJeztah, that's good to know. After digging around I'm convinced the specific server I've been given is horribly misconfigured so it is good to know it works for you.

**Comment by thaJeztah** [comment]
Is it safe to say this is resolved, and related to a server missing some repos that (I think?) should be enabled by default?

**Comment by mikekuriger** [comment]
I also cannot install docker-engine, I'm on CentOS 7.1

# yum install docker-engine
Loaded plugins: fastestmirror, langpacks
Loading mirror speeds from cached hostfile
Resolving Dependencies
--> Running transaction check
---> Package docker-engine.x86_64 0:1.10.2-1.el7.centos will be installed
--> Processing Dependency: libdevmapper.so.1.02(DM_1_02_97)(64bit) for package: docker-engine-1.10.2-1.el7.centos.x86_64
--> Processing Dependency: libsystemd.so.0(LIBSYSTEMD_209)(64bit) for package: docker-engine-1.10.2-1.el7.centos.x86_64
--> Processing Dependency: libsystemd.so.0()(64bit) for package: docker-engine-1.10.2-1.el7.centos.x86_64
--> Finished Dependency Resolution
Error: Package: docker-engine-1.10.2-1.el7.centos.x86_64 (testing)
           Requires: libsystemd.so.0()(64bit)
Error: Package: docker-engine-1.10.2-1.el7.centos.x86_64 (testing)
           Requires: libsystemd.so.0(LIBSYSTEMD_209)(64bit)
Error: Package: docker-engine-1.10.2-1.el7.centos.x86_64 (testing)
           Requires: libdevmapper.so.1.02(DM_1_02_97)(64bit)
 You could try using --skip-broken to work around the problem
 You could try running: rpm -Va --nofiles --nodigest

**Comment by mikekuriger** [comment]
update - upgrading to centos7.2 resolved my issue.
# yum install docker-engine
Loaded plugins: fastestmirror, langpacks
Loading mirror speeds from cached hostfile
Resolving Dependencies
--> Running transaction check
---> Package docker-engine.x86_64 0:1.10.2-1.el7.centos will be installed
--> Finished Dependency Resolution

Dependencies Resolved
Total download size: 9.6 M
Installed size: 41 M
Is this ok [y/d/N]: y
Downloading packages:
docker-engine-1.10.2-1.el7.centos.x86_64.rpm                                    | 9.6 MB  00:00:00     
Running transaction check
Running transaction test
Transaction test succeeded
Running transaction
  Installing : docker-engine-1.10.2-1.el7.centos.x86_64                                            1/1 
  Verifying  : docker-engine-1.10.2-1.el7.centos.x86_64                                            1/1 

Installed:
  docker-engine.x86_64 0:1.10.2-1.el7.centos                                                           

Complete!

**Comment by immarvin** [comment]
hi @thaJeztah , I followed the steps in https://docs.docker.com/engine/installation/linux/rhel/ to install docker engine. It just mentions:
````
Prerequisites

Docker requires a 64-bit installation regardless of your Red Hat version. Docker requires that your kernel must be 3.10 at minimum, which Red Hat 7 runs.

To check your current kernel version, open a terminal and use uname -r to display your kernel version:

$ uname -r
3.10.0-229.el7.x86_64

Finally, is it recommended that you fully update your system. Please keep in mind that your system should be fully patched to fix any potential kernel bugs. Any reported kernel bugs may have already been fixed on the latest kernel packages.
````

I think if docker-engine installation requires some extra redhat repos enabled, it would better be pointed out explicitly in this doc.

**Comment by thaJeztah** [comment]
@immarvin I'm okay with adding a note to that section, just wondering if it is enabled by default already (I.e., if it is enabled by default, should we mention it?). Happy to accept a pull-request though

**Comment by immarvin** [comment]
hi @thaJeztah ,

the content of the default redhat.repo after fresh installation is like this:
````
[root@c910f05c01bc07 ~]# cat /etc/yum.repos.d/redhat.repo 
#
# Certificate-Based Repositories
# Managed by (rhsm) subscription-manager
#
# *** This file is auto-generated.  Changes made here will be over-written. ***
# *** Use "subscription-manager repo-override --help" if you wish to make changes. ***
#
# If this file is empty and this system is subscribed consider 
# a "yum repolist" to refresh available repos
````

**Comment by runcom** [comment]
@immarvin I suspect that's because you don't have an active subscription, on a stock subscribed rhel7.2 I don't have *-updates repo also. I'm retrying with rhel7.1 not subscribed

**Comment by runcom** [comment]
stock rhel7.1 not subscribed has your same exact issue
I'm still not sure nor aware of rhel71-updates repo...
I'm gonna retry with a stock 7.2, then subscribe both and see what happens

**Comment by runcom** [comment]
alright,after subscribing I was able to yum update && yum upgrade and yum install docker-engine correctly.

not sure what to do here..

**Comment by thaJeztah** [comment]
Thanks for testing @runcom 

@immarvin I'm closing this issue, because without subscribing, the RedHat installation itself is incomplete. How to subscribe/install RedHat is out of scope for our documentation.

**Comment by garethr** [comment]
mmm, I'm seeing this on Centos 7.1, so I'm not sure that it's a specific RHEL subscription issue. Also mentioned by @mikekuriger above.

```
$ facter os.release os.name
os.name => CentOS
os.release => {
  full => "7.1.1503",
  major => "7",
  minor => "1"
}
```

Check I have all the standard repos available and the Docker repo:

```
$ yum repolist
Loaded plugins: fastestmirror
Loading mirror speeds from cached hostfile
repo id                                   repo name                                            status
base                                      CentOS-7 - Base                                      8,652
docker                                    Docker                                                  26
extras                                    CentOS-7 - Extras                                      285
updates                                   CentOS-7 - Updates                                   1,730
repolist: 10,693
```

Confirming I'm running the latest available device-mapper 

```
$ yum update -y -q
$ yum install device-mapper
Loaded plugins: fastestmirror
Loading mirror speeds from cached hostfile
Package 7:device-mapper-1.02.93-3.el7_1.1.x86_64 already installed and latest version
Nothing to do
$ yum info device-mapper
Loaded plugins: fastestmirror
Loading mirror speeds from cached hostfile
Installed Packages
Name        : device-mapper
Arch        : x86_64
Epoch       : 7
Version     : 1.02.93
Release     : 3.el7_1.1
Size        : 189 k
Repo        : installed
From repo   : updates
Summary     : Device mapper utility
URL         : http://sources.redhat.com/dm
Licence     : GPLv2
Description : This package contains the supporting userspace utility, dmsetup,
            : for the kernel device-mapper.
```

On trying to install docker-engine 1.10 it fails with the error from the original report.

Other package versions are available in the repos:

```
yum --showduplicates list docker-engine | expand
Loaded plugins: fastestmirror
Loading mirror speeds from cached hostfile
Installed Packages
docker-engine.x86_64                 1.9.1-1.el7.centos                  @docker
Available Packages
docker-engine.x86_64                 1.7.0-1.el7.centos                  docker
docker-engine.x86_64                 1.7.1-1.el7.centos                  docker
docker-engine.x86_64                 1.8.0-1.el7.centos                  docker
docker-engine.x86_64                 1.8.1-1.el7.centos                  docker
docker-engine.x86_64                 1.8.2-1.el7.centos                  docker
docker-engine.x86_64                 1.8.3-1.el7.centos                  docker
docker-engine.x86_64                 1.9.0-1.el7.centos                  docker
docker-engine.x86_64                 1.9.1-1.el7.centos                  docker
docker-engine.x86_64                 1.10.0-1.el7.centos                 docker
docker-engine.x86_64                 1.10.1-1.el7.centos                 docker
docker-engine.x86_64                 1.10.2-1.el7.centos                 docker
```

Trying to install the last 1.9.x release works fine. ie.

```
$ yum install docker-engine-1.9.1-1.el7.centos
```

Will successfully install that version of Docker.

**Comment by thaJeztah** [comment]
@garethr could this be a caching issue, or a mirror not being up to date? (Just thinking out loud); I tried this on a fresh CentOS 7.1 (on DigitalOcean, not bare-metal), and it worked without issues https://github.com/docker/docker/issues/20484#issuecomment-187758636

**Comment by garethr** [comment]
@thaJeztah possibly, I'm talking to the folks who run the mirror in question and will update when I know more.

**Comment by BasLangenberg** [comment]
Just to elaborate on this, I am running into the exact same problems in Oracle Enterprise Linux 7.2. Installing Docker 1.9 works as @garethr points out.

`[root@wmdduymjgcgp9 ~]# yum repolist
Loaded plugins: langpacks, ulninfo
repo id                                                       repo name                                                                                                           status
dockerrepo                                                    Docker Repository                                                                                                       24
ol7_UEKR3/x86_64                                              Latest Unbreakable Enterprise Kernel Release 3 for Oracle Linux 7Server (x86_64)                                       390
ol7_latest/x86_64                                             Oracle Linux 7Server Latest (x86_64)                                                                                13,825
puppetlabs-deps/x86_64                                        Puppet Labs Dependencies El 7 - x86_64                                                                                  17
puppetlabs-products/x86_64                                    Puppet Labs Products El 7 - x86_64                                                                                     205
rhel-7-server-optional-rpms                                   Red Hat Enterprise Linux 7 Server - Optional (RPMs)                                                                  9,007
repolist: 23,468
[root@wmdduymjgcgp9 ~]# yum --showduplicates list docker-engine | expand
Loaded plugins: langpacks, ulninfo
Available Packages
docker-engine.x86_64                   1.8.0-1.el7                    dockerrepo
docker-engine.x86_64                   1.8.1-1.el7                    dockerrepo
docker-engine.x86_64                   1.8.2-1.el7                    dockerrepo
docker-engine.x86_64                   1.8.3-1.el7                    dockerrepo
docker-engine.x86_64                   1.9.0-1.el7                    dockerrepo
docker-engine.x86_64                   1.9.1-1.el7                    dockerrepo
docker-engine.x86_64                   1.10.0-1.el7                   dockerrepo
docker-engine.x86_64                   1.10.1-1.el7                   dockerrepo
docker-engine.x86_64                   1.10.2-1.el7                   dockerrepo
`

**Comment by garethr** [comment]
I'm still tracking this down but it does appear to be a mirror issue. I'll confirm soon all being well.

**Comment by m1n1b00** [comment]
I'm getting the same unresolved dependency errors.  I've followed the instructions on Docker's website to add the yum repo on RHEL 7 and then the docker install fails.  What am I doing wrong?

$ uname -r
3.10.0-229.14.1.el7.x86_64

$ sudo yum install docker-engine
Loaded plugins: langpacks, package_upload, product-id, subscription-manager
dockerrepo                                                                                                                                                    | 2.9 kB  00:00:00
rhel-7-server-extras-rpms                                                                                                                                     | 1.8 kB  00:00:00
rhel-7-server-fastrack-rpms                                                                                                                                   | 1.8 kB  00:00:00
rhel-7-server-optional-fastrack-rpms                                                                                                                          | 1.8 kB  00:00:00
rhel-7-server-optional-rpms                                                                                                                                   | 1.8 kB  00:00:00
rhel-7-server-rh-common-rpms                                                                                                                                  | 2.1 kB  00:00:00
rhel-7-server-rpms                                                                                                                                            | 2.1 kB  00:00:00
rhel-7-server-satellite-tools-6.1-rpms                                                                                                                        | 2.1 kB  00:00:00
rhel-7-server-supplementary-rpms                                                                                                                              | 2.1 kB  00:00:00
rhel-server-rhscl-7-rpms                                                                                                                                      | 2.1 kB  00:00:00
Resolving Dependencies
--> Running transaction check
---> Package docker-engine.x86_64 0:1.10.3-1.el7.centos will be installed
--> Processing Dependency: docker-engine-selinux >= 1.10.3-1.el7.centos for package: docker-engine-1.10.3-1.el7.centos.x86_64
--> Processing Dependency: libdevmapper.so.1.02(DM_1_02_97)(64bit) for package: docker-engine-1.10.3-1.el7.centos.x86_64
--> Processing Dependency: libsystemd.so.0(LIBSYSTEMD_209)(64bit) for package: docker-engine-1.10.3-1.el7.centos.x86_64
--> Processing Dependency: libsystemd.so.0()(64bit) for package: docker-engine-1.10.3-1.el7.centos.x86_64
--> Running transaction check
---> Package docker-engine.x86_64 0:1.10.3-1.el7.centos will be installed
--> Processing Dependency: libdevmapper.so.1.02(DM_1_02_97)(64bit) for package: docker-engine-1.10.3-1.el7.centos.x86_64
--> Processing Dependency: libsystemd.so.0(LIBSYSTEMD_209)(64bit) for package: docker-engine-1.10.3-1.el7.centos.x86_64
--> Processing Dependency: libsystemd.so.0()(64bit) for package: docker-engine-1.10.3-1.el7.centos.x86_64
---> Package docker-engine-selinux.noarch 0:1.10.3-1.el7.centos will be installed
--> Finished Dependency Resolution
Error: Package: docker-engine-1.10.3-1.el7.centos.x86_64 (dockerrepo)
           Requires: libsystemd.so.0(LIBSYSTEMD_209)(64bit)
Error: Package: docker-engine-1.10.3-1.el7.centos.x86_64 (dockerrepo)
           Requires: libsystemd.so.0()(64bit)
Error: Package: docker-engine-1.10.3-1.el7.centos.x86_64 (dockerrepo)
           Requires: libdevmapper.so.1.02(DM_1_02_97)(64bit)
 You could try using --skip-broken to work around the problem
 You could try running: rpm -Va --nofiles --nodigest

**Comment by josdotso** [comment]
sudo yum makecache fast
    sudo yum -y install device-mapper device-mapper-libs libtalloc libtdb libtevent systemd-libs
    sudo yum -y update device-mapper device-mapper-libs libtalloc libtdb libtevent systemd-libs
    # Now install docker-engine

**Comment by m1n1b00** [comment]
I tried the the commands you provided, but I'm still getting the same errors as before.  If I specify an older version of docker then it works, but then I'm stuck using 1.9.1 instead of the 1.10.X:

$ sudo yum install -y http://yum.dockerproject.org/repo/main/centos/7/Packages/docker-engine-1.9.1-1.el7.centos.x86_64.rpm

**Comment by chrishawkins** [comment]
We were only able to resolve this by upgrading the box in question to RHEL 7.2.

**Comment by vishwaraja** [comment]
I bumped into the same issue with CentOS Linux release 7.1.1503 (Core):

Error: Package: docker-engine-1.10.3-1.el7.centos.x86_64 (docker-main-repo)
           Requires: libsystemd.so.0()(64bit)
Error: Package: docker-engine-1.10.3-1.el7.centos.x86_64 (docker-main-repo)
           Requires: libsystemd.so.0(LIBSYSTEMD_209)(64bit)
Error: Package: docker-engine-1.10.3-1.el7.centos.x86_64 (docker-main-repo)
           Requires: libdevmapper.so.1.02(DM_1_02_97)(64bit)
 You could try using --skip-broken to work around the problem
 You could try running: rpm -Va --nofiles --nodigest

It would be great to have a work around that doesn't involve upgrading to centos 7.2 as I just got this box re-imaged with centos 7.1 to use with docker . Also I think docker should update their docs on this .

**Comment by pescobar** [comment]
I am also having this problem when trying to upgrade docker in a machine running `CentOS Linux release 7.1.1503 (Core)`

**Comment by kencochrane** [comment]
I tried to reproduce on my laptop using virtualbox a ``CentOS 7.1.1503`` image from osboxes.org

I followed the steps exactly. 
```
1. $ sudo yum update 

2. Added the docker yum repo

3. $ sudo yum install docker-engine

4. $ sudo service docker start

5. $ sudo docker run hello-world

6. $ uname
Linux osboxes 3.10.0-229.el7.x86_64 #1 SMP Fri Mar 6 11:36:42 UTC 2-15 x86_64 x86_64 x86_64 GNU/Linux
```

The only thing I didn't try was installing docker 1.9 first, and then upgrading to 1.10, but I don't think that would be an issue. If you think I should do that, let me know and I'll test that as well.

Here is the output, yum update removed due to a lot of noise.
```
[osboxes@osboxes ~]$ sudo tee /etc/yum.repos.d/docker.repo <<-'EOF'
> [dockerrepo]
> name=Docker Repository
> baseurl=https://yum.dockerproject.org/repo/main/centos/$releasever/
> enabled=1
> gpgcheck=1
> gpgkey=https://yum.dockerproject.org/gpg
> EOF
[dockerrepo]
name=Docker Repository
baseurl=https://yum.dockerproject.org/repo/main/centos/$releasever/
enabled=1
gpgcheck=1
gpgkey=https://yum.dockerproject.org/gpg

[root@osboxes ~]# yum install docker-engine
Loaded plugins: fastestmirror, langpacks
dockerrepo                                               | 2.9 kB     00:00     
dockerrepo/7/primary_db                                    |  13 kB   00:00     
Loading mirror speeds from cached hostfile
 * base: linux.cc.lehigh.edu
 * extras: centos.mirrors.wvstateu.edu
 * updates: mirror.metrocast.net
Resolving Dependencies
--> Running transaction check
---> Package docker-engine.x86_64 0:1.10.3-1.el7.centos will be installed
--> Processing Dependency: docker-engine-selinux >= 1.10.3-1.el7.centos for package: docker-engine-1.10.3-1.el7.centos.x86_64
--> Running transaction check
---> Package docker-engine-selinux.noarch 0:1.10.3-1.el7.centos will be installed
--> Finished Dependency Resolution

Dependencies Resolved

================================================================================
 Package                  Arch      Version                 Repository     Size
================================================================================
Installing:
 docker-engine            x86_64    1.10.3-1.el7.centos     dockerrepo    9.6 M
Installing for dependencies:
 docker-engine-selinux    noarch    1.10.3-1.el7.centos     dockerrepo     28 k

Transaction Summary
================================================================================
Install  1 Package (+1 Dependent package)

Total download size: 9.6 M
Installed size: 41 M
Is this ok [y/d/N]: y
Downloading packages:
warning: /var/cache/yum/x86_64/7/dockerrepo/packages/docker-engine-1.10.3-1.el7.centos.x86_64.rpm: Header V4 RSA/SHA512 Signature, key ID 2c52609d: NOKEY
Public key for docker-engine-1.10.3-1.el7.centos.x86_64.rpm is not installed
(1/2): docker-engine-1.10.3-1.el7.centos.x86_64.rpm        | 9.6 MB   00:03     
(2/2): docker-engine-selinux-1.10.3-1.el7.centos.noarch.rp |  28 kB   00:06     
--------------------------------------------------------------------------------
Total                                              1.5 MB/s | 9.6 MB  00:06     
Retrieving key from https://yum.dockerproject.org/gpg
Importing GPG key 0x2C52609D:
 Userid     : "Docker Release Tool (releasedocker) <docker@docker.com>"
 Fingerprint: 5811 8e89 f3a9 1289 7c07 0adb f762 2157 2c52 609d
 From       : https://yum.dockerproject.org/gpg
Is this ok [y/N]: y
Running transaction check
Running transaction test
Transaction test succeeded
Running transaction
  Installing : docker-engine-selinux-1.10.3-1.el7.centos.noarch             1/2 
  Installing : docker-engine-1.10.3-1.el7.centos.x86_64                     2/2 
  Verifying  : docker-engine-selinux-1.10.3-1.el7.centos.noarch             1/2 
  Verifying  : docker-engine-1.10.3-1.el7.centos.x86_64                     2/2 

Installed:
  docker-engine.x86_64 0:1.10.3-1.el7.centos                                    

Dependency Installed:
  docker-engine-selinux.noarch 0:1.10.3-1.el7.centos                            

Complete!

[root@osboxes ~]# service docker start
Redirecting to /bin/systemctl start  docker.service
[root@osboxes ~]# docker ps
CONTAINER ID        IMAGE               COMMAND             CREATED             STATUS              PORTS               NAMES
[root@osboxes ~]# docker run hello-world
Unable to find image 'hello-world:latest' locally
latest: Pulling from library/hello-world
03f4658f8b78: Pull complete 
a3ed95caeb02: Pull complete 
Digest: sha256:8be990ef2aeb16dbcb9271ddfe2610fa6658d13f6dfb8bc72074cc1ca36966a7
Status: Downloaded newer image for hello-world:latest

Hello from Docker.
This message shows that your installation appears to be working correctly.

To generate this message, Docker took the following steps:
 1. The Docker client contacted the Docker daemon.
 2. The Docker daemon pulled the "hello-world" image from the Docker Hub.
 3. The Docker daemon created a new container from that image which runs the
    executable that produces the output you are currently reading.
 4. The Docker daemon streamed that output to the Docker client, which sent it
    to your terminal.

To try something more ambitious, you can run an Ubuntu container with:
 $ docker run -it ubuntu bash

Share images, automate workflows, and more with a free Docker Hub account:
 https://hub.docker.com

For more examples and ideas, visit:
 https://docs.docker.com/userguide/

[root@osboxes ~]# docker version
Client:
 Version:      1.10.3
 API version:  1.22
 Go version:   go1.5.3
 Git commit:   20f81dd
 Built:        Thu Mar 10 15:39:25 2016
 OS/Arch:      linux/amd64

Server:
 Version:      1.10.3
 API version:  1.22
 Go version:   go1.5.3
 Git commit:   20f81dd
 Built:        Thu Mar 10 15:39:25 2016
 OS/Arch:      linux/amd64

[root@osboxes ~]# uname -a
Linux osboxes 3.10.0-229.el7.x86_64 #1 SMP Fri Mar 6 11:36:42 UTC 2015 x86_64 x86_64 x86_64 GNU/Linux
```

**Comment by pescobar** [comment]
@kencochrane can you check the output of `cat /etc/redhat-release` in your test machine?

**Comment by kencochrane** [comment]
@pescobar  `CentOS Linux release 7.2.1511 (Core)`

**Comment by pescobar** [comment]
@kencochrane that's what I supposed and why I asked you to check /etc/redhat-release file. I think when you did `yum update` you moved to 7.2. The issue only appear with 7.1

```
[root@sysmon01 ~]# cat /etc/redhat-release
CentOS Linux release 7.1.1503 (Core)
```

**Comment by kencochrane** [comment]
With a brand new image:

```
$ cat /etc/redhat-release
Derived from Red Hat Enterprise Linux 7.1 (Source)

$ cat /etc/centos-release
CentOS Linux release 7.1.1503 (Core)
```

Ran $ yum update

```
$ cat /etc/redhat-release
CentOS Linux release 7.2.1511 (Core)

$ cat /etc/centos-release
CentOS Linux release 7.2.1511 (Core)
```

So it looks like running yum update, upgrades from 7.1 to 7.2. 

Is there a reason why people don't want to upgrade from Cent OS 7.1 to 7.2? Is it not a safe operation? Sorry I'm more familiar with Debian based systems.

Testing without doing a yum update now.

**Comment by kencochrane** [comment]
@pescobar  OK, ran it again, but this time I didn't do the yum update, and it still worked.

Here is my full output

```
dockerrepo                                                                        | 2.9 kB  00:00:00     
dockerrepo/7/primary_db                                                           |  13 kB  00:00:00     
Loading mirror speeds from cached hostfile
 * base: mirror.cogentco.com
 * extras: centos-mirror.jchost.net
 * updates: bay.uchicago.edu
Resolving Dependencies
--> Running transaction check
---> Package docker-engine.x86_64 0:1.10.3-1.el7.centos will be installed
--> Processing Dependency: docker-engine-selinux >= 1.10.3-1.el7.centos for package: docker-engine-1.10.3-1.el7.centos.x86_64
--> Processing Dependency: libdevmapper.so.1.02(DM_1_02_97)(64bit) for package: docker-engine-1.10.3-1.el7.centos.x86_64
--> Processing Dependency: libsystemd.so.0(LIBSYSTEMD_209)(64bit) for package: docker-engine-1.10.3-1.el7.centos.x86_64
--> Processing Dependency: libsystemd.so.0()(64bit) for package: docker-engine-1.10.3-1.el7.centos.x86_64
--> Running transaction check
---> Package device-mapper-libs.x86_64 7:1.02.93-3.el7 will be updated
--> Processing Dependency: device-mapper-libs = 7:1.02.93-3.el7 for package: 7:device-mapper-1.02.93-3.el7.x86_64
---> Package device-mapper-libs.x86_64 7:1.02.107-5.el7_2.1 will be an update
---> Package docker-engine-selinux.noarch 0:1.10.3-1.el7.centos will be installed
---> Package systemd-libs.x86_64 0:208-20.el7 will be updated
--> Processing Dependency: systemd-libs = 208-20.el7 for package: systemd-208-20.el7.x86_64
---> Package systemd-libs.x86_64 0:219-19.el7_2.7 will be an update
--> Running transaction check
---> Package device-mapper.x86_64 7:1.02.93-3.el7 will be updated
--> Processing Dependency: device-mapper = 7:1.02.93-3.el7 for package: 7:device-mapper-event-1.02.93-3.el7.x86_64
---> Package device-mapper.x86_64 7:1.02.107-5.el7_2.1 will be an update
---> Package systemd.x86_64 0:208-20.el7 will be updated
--> Processing Dependency: systemd = 208-20.el7 for package: systemd-python-208-20.el7.x86_64
--> Processing Dependency: systemd = 208-20.el7 for package: systemd-sysv-208-20.el7.x86_64
--> Processing Dependency: systemd = 208-20.el7 for package: libgudev1-208-20.el7.x86_64
---> Package systemd.x86_64 0:219-19.el7_2.7 will be an update
--> Processing Dependency: kmod >= 18-4 for package: systemd-219-19.el7_2.7.x86_64
--> Running transaction check
---> Package device-mapper-event.x86_64 7:1.02.93-3.el7 will be updated
--> Processing Dependency: device-mapper-event = 7:1.02.93-3.el7 for package: 7:lvm2-libs-2.02.115-3.el7.x86_64
---> Package device-mapper-event.x86_64 7:1.02.107-5.el7_2.1 will be an update
--> Processing Dependency: device-mapper-event-libs = 7:1.02.107-5.el7_2.1 for package: 7:device-mapper-event-1.02.107-5.el7_2.1.x86_64
---> Package kmod.x86_64 0:14-10.el7 will be updated
---> Package kmod.x86_64 0:20-5.el7 will be an update
---> Package libgudev1.x86_64 0:208-20.el7 will be updated
---> Package libgudev1.x86_64 0:219-19.el7_2.7 will be an update
---> Package systemd-python.x86_64 0:208-20.el7 will be updated
---> Package systemd-python.x86_64 0:219-19.el7_2.7 will be an update
---> Package systemd-sysv.x86_64 0:208-20.el7 will be updated
---> Package systemd-sysv.x86_64 0:219-19.el7_2.7 will be an update
--> Running transaction check
---> Package device-mapper-event-libs.x86_64 7:1.02.93-3.el7 will be updated
---> Package device-mapper-event-libs.x86_64 7:1.02.107-5.el7_2.1 will be an update
---> Package lvm2-libs.x86_64 7:2.02.115-3.el7 will be updated
--> Processing Dependency: lvm2-libs = 7:2.02.115-3.el7 for package: 7:lvm2-2.02.115-3.el7.x86_64
---> Package lvm2-libs.x86_64 7:2.02.130-5.el7_2.1 will be an update
--> Running transaction check
---> Package lvm2.x86_64 7:2.02.115-3.el7 will be updated
---> Package lvm2.x86_64 7:2.02.130-5.el7_2.1 will be an update
--> Processing Dependency: device-mapper-persistent-data >= 0.5.5-1 for package: 7:lvm2-2.02.130-5.el7_2.1.x86_64
--> Running transaction check
---> Package device-mapper-persistent-data.x86_64 0:0.4.1-2.el7 will be updated
---> Package device-mapper-persistent-data.x86_64 0:0.5.5-1.el7 will be an update
--> Processing Conflict: systemd-219-19.el7_2.7.x86_64 conflicts initscripts < 9.49.28-1
--> Restarting Dependency Resolution with new changes.
--> Running transaction check
---> Package initscripts.x86_64 0:9.49.24-1.el7 will be updated
---> Package initscripts.x86_64 0:9.49.30-1.el7_2.2 will be an update
--> Processing Conflict: systemd-219-19.el7_2.7.x86_64 conflicts dracut < 033-243
--> Restarting Dependency Resolution with new changes.
--> Running transaction check
---> Package dracut.x86_64 0:033-240.el7 will be updated
--> Processing Dependency: dracut = 033-240.el7 for package: dracut-config-rescue-033-240.el7.x86_64
--> Processing Dependency: dracut = 033-240.el7 for package: dracut-network-033-240.el7.x86_64
---> Package dracut.x86_64 0:033-360.el7_2 will be an update
--> Running transaction check
---> Package dracut-config-rescue.x86_64 0:033-240.el7 will be updated
---> Package dracut-config-rescue.x86_64 0:033-360.el7_2 will be an update
---> Package dracut-network.x86_64 0:033-240.el7 will be updated
---> Package dracut-network.x86_64 0:033-360.el7_2 will be an update
--> Finished Dependency Resolution

Dependencies Resolved

=========================================================================================================
 Package                              Arch          Version                      Repository         Size
=========================================================================================================
Installing:
 docker-engine                        x86_64        1.10.3-1.el7.centos          dockerrepo        9.6 M
Updating:
 dracut                               x86_64        033-360.el7_2                updates           311 k
 initscripts                          x86_64        9.49.30-1.el7_2.2            updates           429 k
Installing for dependencies:
 docker-engine-selinux                noarch        1.10.3-1.el7.centos          dockerrepo         28 k
Updating for dependencies:
 device-mapper                        x86_64        7:1.02.107-5.el7_2.1         updates           252 k
 device-mapper-event                  x86_64        7:1.02.107-5.el7_2.1         updates           167 k
 device-mapper-event-libs             x86_64        7:1.02.107-5.el7_2.1         updates           169 k
 device-mapper-libs                   x86_64        7:1.02.107-5.el7_2.1         updates           304 k
 device-mapper-persistent-data        x86_64        0.5.5-1.el7                  base              350 k
 dracut-config-rescue                 x86_64        033-360.el7_2                updates            49 k
 dracut-network                       x86_64        033-360.el7_2                updates            90 k
 kmod                                 x86_64        20-5.el7                     base              114 k
 libgudev1                            x86_64        219-19.el7_2.7               updates            65 k
 lvm2                                 x86_64        7:2.02.130-5.el7_2.1         updates           1.0 M
 lvm2-libs                            x86_64        7:2.02.130-5.el7_2.1         updates           872 k
 systemd                              x86_64        219-19.el7_2.7               updates           5.1 M
 systemd-libs                         x86_64        219-19.el7_2.7               updates           357 k
 systemd-python                       x86_64        219-19.el7_2.7               updates            99 k
 systemd-sysv                         x86_64        219-19.el7_2.7               updates            53 k

Transaction Summary
=========================================================================================================
Install  1 Package  (+ 1 Dependent package)
Upgrade  2 Packages (+15 Dependent packages)

Total size: 19 M
Total download size: 9.6 M
Is this ok [y/d/N]: y
Downloading packages:
warning: /var/cache/yum/x86_64/7/dockerrepo/packages/docker-engine-1.10.3-1.el7.centos.x86_64.rpm: Header V4 RSA/SHA512 Signature, key ID 2c52609d: NOKEY
Public key for docker-engine-1.10.3-1.el7.centos.x86_64.rpm is not installed
(1/2): docker-engine-1.10.3-1.el7.centos.x86_64.rpm                               | 9.6 MB  00:00:03     
(2/2): docker-engine-selinux-1.10.3-1.el7.centos.noarch.rpm                       |  28 kB  00:00:06     
---------------------------------------------------------------------------------------------------------
Total                                                                    1.5 MB/s | 9.6 MB  00:00:06     
warning: /var/cache/yum/x86_64/7/updates/packages/initscripts-9.49.30-1.el7_2.2.x86_64.rpm: Header V3 RSA/SHA256 Signature, key ID f4a80eb5: NOKEY
Retrieving key from file:///etc/pki/rpm-gpg/RPM-GPG-KEY-CentOS-7
Importing GPG key 0xF4A80EB5:
 Userid     : "CentOS-7 Key (CentOS 7 Official Signing Key) <security@centos.org>"
 Fingerprint: 6341 ab27 53d7 8a78 a7c2 7bb1 24c6 a8a7 f4a8 0eb5
 Package    : centos-release-7-1.1503.el7.centos.2.7.x86_64 (@base/$releasever)
 From       : /etc/pki/rpm-gpg/RPM-GPG-KEY-CentOS-7
Is this ok [y/N]: y
Retrieving key from https://yum.dockerproject.org/gpg
Importing GPG key 0x2C52609D:
 Userid     : "Docker Release Tool (releasedocker) <docker@docker.com>"
 Fingerprint: 5811 8e89 f3a9 1289 7c07 0adb f762 2157 2c52 609d
 From       : https://yum.dockerproject.org/gpg
Is this ok [y/N]: y
Running transaction check
Running transaction test
Transaction test succeeded
Running transaction
  Updating   : systemd-libs-219-19.el7_2.7.x86_64                                                   1/36 
  Updating   : systemd-219-19.el7_2.7.x86_64                                                        2/36 
  Updating   : dracut-033-360.el7_2.x86_64                                                          3/36 
  Updating   : kmod-20-5.el7.x86_64                                                                 4/36 
  Updating   : 7:device-mapper-libs-1.02.107-5.el7_2.1.x86_64                                       5/36 
  Updating   : 7:device-mapper-1.02.107-5.el7_2.1.x86_64                                            6/36 
  Updating   : 7:device-mapper-event-libs-1.02.107-5.el7_2.1.x86_64                                 7/36 
  Updating   : 7:device-mapper-event-1.02.107-5.el7_2.1.x86_64                                      8/36 
  Updating   : 7:lvm2-libs-2.02.130-5.el7_2.1.x86_64                                                9/36 
  Updating   : device-mapper-persistent-data-0.5.5-1.el7.x86_64                                    10/36 
  Installing : docker-engine-selinux-1.10.3-1.el7.centos.noarch                                    11/36 
  Installing : docker-engine-1.10.3-1.el7.centos.x86_64                                            12/36 
  Updating   : 7:lvm2-2.02.130-5.el7_2.1.x86_64                                                    13/36 
Created symlink from /etc/systemd/system/sysinit.target.wants/lvm2-lvmpolld.socket to /usr/lib/systemd/system/lvm2-lvmpolld.socket.
  Updating   : initscripts-9.49.30-1.el7_2.2.x86_64                                                14/36 
  Updating   : dracut-network-033-360.el7_2.x86_64                                                 15/36 
  Updating   : dracut-config-rescue-033-360.el7_2.x86_64                                           16/36 
  Updating   : systemd-python-219-19.el7_2.7.x86_64                                                17/36 
  Updating   : systemd-sysv-219-19.el7_2.7.x86_64                                                  18/36 
  Updating   : libgudev1-219-19.el7_2.7.x86_64                                                     19/36 
  Cleanup    : 7:lvm2-2.02.115-3.el7.x86_64                                                        20/36 
  Cleanup    : 7:lvm2-libs-2.02.115-3.el7.x86_64                                                   21/36 
  Cleanup    : 7:device-mapper-event-1.02.93-3.el7.x86_64                                          22/36 
  Cleanup    : systemd-python-208-20.el7.x86_64                                                    23/36 
  Cleanup    : libgudev1-208-20.el7.x86_64                                                         24/36 
  Cleanup    : initscripts-9.49.24-1.el7.x86_64                                                    25/36 
  Cleanup    : systemd-sysv-208-20.el7.x86_64                                                      26/36 
  Cleanup    : dracut-config-rescue-033-240.el7.x86_64                                             27/36 
  Cleanup    : dracut-network-033-240.el7.x86_64                                                   28/36 
  Cleanup    : dracut-033-240.el7.x86_64                                                           29/36 
  Cleanup    : 7:device-mapper-event-libs-1.02.93-3.el7.x86_64                                     30/36 
  Cleanup    : 7:device-mapper-1.02.93-3.el7.x86_64                                                31/36 
  Cleanup    : 7:device-mapper-libs-1.02.93-3.el7.x86_64                                           32/36 
  Cleanup    : systemd-208-20.el7.x86_64                                                           33/36 
  Cleanup    : kmod-14-10.el7.x86_64                                                               34/36 
  Cleanup    : systemd-libs-208-20.el7.x86_64                                                      35/36 
  Cleanup    : device-mapper-persistent-data-0.4.1-2.el7.x86_64                                    36/36 
  Verifying  : initscripts-9.49.30-1.el7_2.2.x86_64                                                 1/36 
  Verifying  : 7:device-mapper-event-libs-1.02.107-5.el7_2.1.x86_64                                 2/36 
  Verifying  : docker-engine-selinux-1.10.3-1.el7.centos.noarch                                     3/36 
  Verifying  : libgudev1-219-19.el7_2.7.x86_64                                                      4/36 
  Verifying  : dracut-network-033-360.el7_2.x86_64                                                  5/36 
  Verifying  : kmod-20-5.el7.x86_64                                                                 6/36 
  Verifying  : systemd-219-19.el7_2.7.x86_64                                                        7/36 
  Verifying  : 7:device-mapper-event-1.02.107-5.el7_2.1.x86_64                                      8/36 
  Verifying  : dracut-config-rescue-033-360.el7_2.x86_64                                            9/36 
  Verifying  : 7:device-mapper-libs-1.02.107-5.el7_2.1.x86_64                                      10/36 
  Verifying  : 7:lvm2-libs-2.02.130-5.el7_2.1.x86_64                                               11/36 
  Verifying  : systemd-python-219-19.el7_2.7.x86_64                                                12/36 
  Verifying  : 7:device-mapper-1.02.107-5.el7_2.1.x86_64                                           13/36 
  Verifying  : dracut-033-360.el7_2.x86_64                                                         14/36 
  Verifying  : systemd-libs-219-19.el7_2.7.x86_64                                                  15/36 
  Verifying  : device-mapper-persistent-data-0.5.5-1.el7.x86_64                                    16/36 
  Verifying  : systemd-sysv-219-19.el7_2.7.x86_64                                                  17/36 
  Verifying  : 7:lvm2-2.02.130-5.el7_2.1.x86_64                                                    18/36 
  Verifying  : docker-engine-1.10.3-1.el7.centos.x86_64                                            19/36 
  Verifying  : 7:device-mapper-event-1.02.93-3.el7.x86_64                                          20/36 
  Verifying  : 7:device-mapper-event-libs-1.02.93-3.el7.x86_64                                     21/36 
  Verifying  : dracut-network-033-240.el7.x86_64                                                   22/36 
  Verifying  : dracut-033-240.el7.x86_64                                                           23/36 
  Verifying  : kmod-14-10.el7.x86_64                                                               24/36 
  Verifying  : libgudev1-208-20.el7.x86_64                                                         25/36 
  Verifying  : 7:device-mapper-libs-1.02.93-3.el7.x86_64                                           26/36 
  Verifying  : 7:lvm2-2.02.115-3.el7.x86_64                                                        27/36 
  Verifying  : systemd-208-20.el7.x86_64                                                           28/36 
  Verifying  : initscripts-9.49.24-1.el7.x86_64                                                    29/36 
  Verifying  : systemd-libs-208-20.el7.x86_64                                                      30/36 
  Verifying  : 7:device-mapper-1.02.93-3.el7.x86_64                                                31/36 
  Verifying  : 7:lvm2-libs-2.02.115-3.el7.x86_64                                                   32/36 
  Verifying  : dracut-config-rescue-033-240.el7.x86_64                                             33/36 
  Verifying  : systemd-sysv-208-20.el7.x86_64                                                      34/36 
  Verifying  : systemd-python-208-20.el7.x86_64                                                    35/36 
  Verifying  : device-mapper-persistent-data-0.4.1-2.el7.x86_64                                    36/36 

Installed:
  docker-engine.x86_64 0:1.10.3-1.el7.centos                                                             

Dependency Installed:
  docker-engine-selinux.noarch 0:1.10.3-1.el7.centos                                                     

Updated:
  dracut.x86_64 0:033-360.el7_2                  initscripts.x86_64 0:9.49.30-1.el7_2.2                 

Dependency Updated:
  device-mapper.x86_64 7:1.02.107-5.el7_2.1             device-mapper-event.x86_64 7:1.02.107-5.el7_2.1 
  device-mapper-event-libs.x86_64 7:1.02.107-5.el7_2.1  device-mapper-libs.x86_64 7:1.02.107-5.el7_2.1  
  device-mapper-persistent-data.x86_64 0:0.5.5-1.el7    dracut-config-rescue.x86_64 0:033-360.el7_2     
  dracut-network.x86_64 0:033-360.el7_2                 kmod.x86_64 0:20-5.el7                          
  libgudev1.x86_64 0:219-19.el7_2.7                     lvm2.x86_64 7:2.02.130-5.el7_2.1                
  lvm2-libs.x86_64 7:2.02.130-5.el7_2.1                 systemd.x86_64 0:219-19.el7_2.7                 
  systemd-libs.x86_64 0:219-19.el7_2.7                  systemd-python.x86_64 0:219-19.el7_2.7          
  systemd-sysv.x86_64 0:219-19.el7_2.7                 

Complete!
[root@osboxes ~]# service docker start
Redirecting to /bin/systemctl start  docker.service
[root@osboxes ~]# docker ps
CONTAINER ID        IMAGE               COMMAND             CREATED             STATUS              PORTS               NAMES
[root@osboxes ~]# docker run hello-world
Unable to find image 'hello-world:latest' locally
latest: Pulling from library/hello-world
03f4658f8b78: Pull complete 
a3ed95caeb02: Pull complete 
Digest: sha256:8be990ef2aeb16dbcb9271ddfe2610fa6658d13f6dfb8bc72074cc1ca36966a7
Status: Downloaded newer image for hello-world:latest

Hello from Docker.
This message shows that your installation appears to be working correctly.

To generate this message, Docker took the following steps:
 1. The Docker client contacted the Docker daemon.
 2. The Docker daemon pulled the "hello-world" image from the Docker Hub.
 3. The Docker daemon created a new container from that image which runs the
    executable that produces the output you are currently reading.
 4. The Docker daemon streamed that output to the Docker client, which sent it
    to your terminal.

To try something more ambitious, you can run an Ubuntu container with:
 $ docker run -it ubuntu bash

Share images, automate workflows, and more with a free Docker Hub account:
 https://hub.docker.com

For more examples and ideas, visit:
 https://docs.docker.com/userguide/


$ cat /etc/redhat-release 
Derived from Red Hat Enterprise Linux 7.1 (Source)

$ cat /etc/centos-release 
CentOS Linux release 7.1.1503 (Core)
```

**Comment by pescobar** [comment]
thanks for the tests @kencochrane 

In my specific case this machine is connected to a infiniband network and is a gpfs client. Usually when a new centos version is released the infiniband driver and gpfs client takes some time to be supported in the new centos version. That's why we have an internal centos mirror and we modify the yum repos config to "hardcode" the centos version. This way we avoid automatic upgrades to newer centos versions until we are sure the upgrade is supported.

I think the docker installation is working for you because you are getting `systemd-libs.x86_64 0:219-19.el7_2.7` rpm from centos 7.2 repository while in my local mirror I have the centos 7.1 rpm which is `systemd-libs-208-20.el7_1.5.x86_64`

```
[root@sysmon01 ~]# yum install systemd-libs
Package systemd-libs-208-20.el7_1.5.x86_64 already installed and latest version
Nothing to do
```

Personally I will test if I can update this machine to latest centos version. As centos 7.2 was releases few months ago I think it should work but maybe there is other people in similar situations who cannot upgrade.

**Comment by kencochrane** [comment]
@pescobar ok, thanks for the details, it makes sense. 

I think you are right, if you look at the version I have installed systemd-libs.x86_64 0:219-19.**el7_2**.7 it is `el7_2` vs yours which is `el7_1`

**Comment by kkorada** [comment]
**when i am trying to install docker on RHEL 7 i am getting following error**
Resolving Dependencies
--> Running transaction check
---> Package docker-engine.x86_64 0:1.10.3-1.el7.centos will be installed
--> Processing Dependency: selinux-policy >= 3.13.1-23 for package: docker-engine-1.10.3-1.el7.centos.x86_64
--> Processing Dependency: docker-engine-selinux >= 1.10.3-1.el7.centos for package: docker-engine-1.10.3-1.el7.centos.x86_64
--> Processing Dependency: libdevmapper.so.1.02(DM_1_02_97)(64bit) for package: docker-engine-1.10.3-1.el7.centos.x86_64
--> Processing Dependency: libsystemd.so.0(LIBSYSTEMD_209)(64bit) for package: docker-engine-1.10.3-1.el7.centos.x86_64
--> Processing Dependency: libsystemd.so.0()(64bit) for package: docker-engine-1.10.3-1.el7.centos.x86_64
--> Running transaction check
---> Package docker-engine.x86_64 0:1.10.3-1.el7.centos will be installed
--> Processing Dependency: selinux-policy >= 3.13.1-23 for package: docker-engine-1.10.3-1.el7.centos.x86_64
--> Processing Dependency: libdevmapper.so.1.02(DM_1_02_97)(64bit) for package: docker-engine-1.10.3-1.el7.centos.x86_64
--> Processing Dependency: libsystemd.so.0(LIBSYSTEMD_209)(64bit) for package: docker-engine-1.10.3-1.el7.centos.x86_64
--> Processing Dependency: libsystemd.so.0()(64bit) for package: docker-engine-1.10.3-1.el7.centos.x86_64
---> Package docker-engine-selinux.noarch 0:1.10.3-1.el7.centos will be installed
--> Processing Dependency: selinux-policy-base >= 3.13.1-23 for package: docker-engine-selinux-1.10.3-1.el7.centos.noarch
--> Processing Dependency: selinux-policy-targeted >= 3.13.1-23 for package: docker-engine-selinux-1.10.3-1.el7.centos.noarch
--> Processing Dependency: policycoreutils-python for package: docker-engine-selinux-1.10.3-1.el7.centos.noarch
--> Finished Dependency Resolution
Error: Package: docker-engine-selinux-1.10.3-1.el7.centos.noarch (dockerrepo)
           Requires: policycoreutils-python
Error: Package: docker-engine-selinux-1.10.3-1.el7.centos.noarch (dockerrepo)
           Requires: selinux-policy-base >= 3.13.1-23
           Installed: selinux-policy-targeted-3.12.1-153.el7.noarch (@anaconda/7.0)
               selinux-policy-base = 3.12.1-153.el7
Error: Package: docker-engine-1.10.3-1.el7.centos.x86_64 (dockerrepo)
           Requires: libsystemd.so.0(LIBSYSTEMD_209)(64bit)
Error: Package: docker-engine-1.10.3-1.el7.centos.x86_64 (dockerrepo)
           Requires: libsystemd.so.0()(64bit)
Error: Package: docker-engine-selinux-1.10.3-1.el7.centos.noarch (dockerrepo)
           Requires: selinux-policy-targeted >= 3.13.1-23
           Installed: selinux-policy-targeted-3.12.1-153.el7.noarch (@anaconda/7.0)
               selinux-policy-targeted = 3.12.1-153.el7
Error: Package: docker-engine-1.10.3-1.el7.centos.x86_64 (dockerrepo)
           Requires: selinux-policy >= 3.13.1-23
           Installed: selinux-policy-3.12.1-153.el7.noarch (@anaconda/7.0)
               selinux-policy = 3.12.1-153.el7
Error: Package: docker-engine-1.10.3-1.el7.centos.x86_64 (dockerrepo)
           Requires: libdevmapper.so.1.02(DM_1_02_97)(64bit)
 You could try using --skip-broken to work around the problem
 You could try running: rpm -Va --nofiles --nodigest

**Comment by kencochrane** [comment]
@kkorada did you run yum update before you tried to install docker? It says your systemd lib version is too old, and needs to be upgraded.

``Requires: libsystemd.so.0(LIBSYSTEMD_209)(64bit)``

**Comment by kkorada** [comment]
yes i ran yum update before doing this. but getting same error

**Comment by ali-himindz** [comment]
@kkorada Same issue for me, on RHEL 7.0 with kernel 3.10.0-123.el7.x86_64. Lets if it can be solved today :).

**Comment by kkorada** [comment]
yes lets see! I am also using RHEL 7.0 with kernel 3.10.0-123.el7.x86_64

**Comment by ali-himindz** [comment]
Ok I managed to install it by adding a repo in /etc/yum.repos.d/centos7.repo using the baseurl http://mirror.centos.org/centos/7/os/x86_64/
then I did the 
sudo yum install docker-engine
And it was able to find the dependencies from this repo. Please note on my machine did not have RHEL updates repo via subscription manager, therefore I had to use the centos7 repo.

**Comment by thaJeztah** [comment]
@ali-himindz since you mentioned you're running RHEL7, and you fixed it by adding a CentOS repository; do you have an active subscription? See https://github.com/docker/docker/issues/20484#issuecomment-188648599 (and the comments below that). If you don't have an active subscription, you won't be able to get updates from the Red Hat repositories, in which case this is expected.

**Comment by ali-himindz** [comment]
Exactly this is what I found. I didn't need the centos7 report instead I
just activated the subscription via subscription-manager.
Thanks
Ali

**Comment by kkorada** [comment]
@ali-himindz i dont have an active subscription, can you please tell me the workaround you followed my adding centos7.repo ? and attach you centos7.repo

**Comment by kkorada** [comment]
@ali-himindz  can i use centos7 repo for installing docker without a subscription for RHEL7 extra rpms ?

**Comment by ali-himindz** [comment]
Hi @thaJeztah sorry I was away. Yes Centos7 and RHEL7 are binary-compatible, I have used the CentOS7 RPMS on RHEL and vice versa. As far as the licensing terms are concerned please check with you company policy and your existing license of RHEL.

**Comment by TomBarryy** [comment]
Hi,

I am having similar issues as previously mentioned:

`cat /etc/redhat-release`
`Red Hat Enterprise Linux Server release 7.2 (Maipo)`

`uname -r`
`3.10.0-123.el7.x86_64`

Error Message:
`yum install docker-engine`
`Loaded plugins: langpacks, product-id, rhnplugin, search-disabled-repos`
`This system is receiving updates from RHN Classic or Red Hat Satellite.`
`Resolving Dependencies`
`--> Running transaction check`
`---> Package docker-engine.x86_64 0:1.11.2-1.el7.centos will be installed`
`--> Processing Dependency: docker-engine-selinux >= 1.11.2-1.el7.centos for package: docker-engine-1.11.2-1.el7.centos.x86_64`
`--> Running transaction check`
`---> Package docker-engine-selinux.noarch 0:1.11.2-1.el7.centos will be installed`
`--> Processing Conflict: docker-engine-1.11.2-1.el7.centos.x86_64 conflicts docker`
`--> Finished Dependency Resolution`
`Error: docker-engine conflicts with docker-1.1.2-13.el7.x86_64`
` You could try using --skip-broken to work around the problem`
` You could try running: rpm -Va --nofiles --nodigest`

I have subscribed version of RHEL.

Unsure of how to proceed. Any advice?

**Comment by thaJeztah** [comment]
@TomBarryy looks like you have an old docker 1.1.2 package installed from the Red Hat repository;

> Error: docker-engine conflicts with docker-1.1.2-13.el7.x86_64

Uninstall the existing `docker` package, then try installing docker-engine again

**Comment by TomBarryy** [comment]
Thank you @thaJeztah, that was all that was required!

**Comment by balamsg** [comment]
Hi,
    I am a newbie in this docker environment.So, can somebody tell me how to install docker without yum in RHEL 7.1 .

**Comment by balamsg** [comment]
Hi,
I am a newbie in this docker environment.So, can somebody tell me how to install docker without yum in RHEL 7.1 . Not able to get much info abt this in the google search. I have downloaded "docker-engine-1.10.0-1.el7.centos.x86_64.rpm" and "docker-engine-selinux-1.10.0-1.el7.centos.noarch.rpm" but still it says

**Comment by balamsg** [comment]
Hi,
I am a newbie in this docker environment.So, can somebody tell me how to install docker without yum in RHEL 7.1 . Not able to get much info abt this in the google search. I have downloaded "docker-engine-1.10.0-1.el7.centos.x86_64.rpm" and "docker-engine-selinux-1.10.0-1.el7.centos.noarch.rpm" but still it says                                                                                    

warning: docker-engine-1.10.0-1.el7.centos.x86_64.rpm: Header V4 RSA/SHA512 Signature, key ID 2c52609d: NOKEY
error: Failed dependencies:
        libdevmapper.so.1.02(DM_1_02_97)(64bit) is needed by docker-engine-1.10.0-1.el7.centos.x86_64
        libsystemd.so.0()(64bit) is needed by docker-engine-1.10.0-1.el7.centos.x86_64
        libsystemd.so.0(LIBSYSTEMD_209)(64bit) is needed by docker-engine-1.10.0-1.el7.centos.x86_64

**Comment by rkgade** [comment]
Hi @balamsg,

If you are at Cent OS 7.1, you might have to upgrade to 7.2.
Missing .so files are updated/installed when you do a yum update.

To install docker you might not need yum, but to download dependencies on
Cent OS, you need to do a yum update.

**Comment by systemroot** [comment]
**`yum install docker-engine-1.9.1-1.el7.centos`**
`Package matching docker-engine-1.9.1-1.el7.centos.x86_64 already installed. Checking for update.
Nothing to do`

**`docker -v`**
`Docker version 1.11.2, build b9f10c9`

**`uname -a`**
<pre>
LSB Version:	:core-4.1-amd64:core-4.1-noarch
Distributor ID:	CentOS
Description:	CentOS Linux release 7.2.1511 (Core) 
Release:	7.2.1511
Codename:	Core</pre>

<pre>
yum list installed | grep docker
docker-engine.x86_64              1.11.2-1.el7.centos                @docker-main-repo
docker-engine-selinux.noarch      1.11.2-1.el7.centos                @dockerrepo
</pre>

How can i install docker 1.9 ? Please Help!

**Comment by terryzwt** [comment]
Fix this by update the Centos to 7.3

**Comment by vikaschenny** [comment]
am trying to install docker-ee-stable-17.03 am getting error can any one help me 
os -- Red Hat Enterprise Linux Server release 7.2 (Maipo)



[root@dockerslaveone yum.repos.d]# yum install docker-ee
Loaded plugins: langpacks, product-id, search-disabled-repos, subscription-manager
This system is not registered to Red Hat Subscription Management. You can use subscription-manager to register.
docker-ee-stable-17.03                                                                                                                                                   | 2.9 kB  00:00:00
Resolving Dependencies
--> Running transaction check
---> Package docker-ee.x86_64 0:17.03.2.ee.4-1.el7.centos will be installed
--> Processing Dependency: docker-ee-selinux >= 17.03.2.ee.4-1.el7.centos for package: docker-ee-17.03.2.ee.4-1.el7.centos.x86_64
--> Processing Dependency: libltdl.so.7()(64bit) for package: docker-ee-17.03.2.ee.4-1.el7.centos.x86_64
--> Processing Dependency: libseccomp.so.2()(64bit) for package: docker-ee-17.03.2.ee.4-1.el7.centos.x86_64
--> Running transaction check
---> Package docker-ee.x86_64 0:17.03.2.ee.4-1.el7.centos will be installed
--> Processing Dependency: libltdl.so.7()(64bit) for package: docker-ee-17.03.2.ee.4-1.el7.centos.x86_64
--> Processing Dependency: libseccomp.so.2()(64bit) for package: docker-ee-17.03.2.ee.4-1.el7.centos.x86_64
---> Package docker-ee-selinux.noarch 0:17.03.2.ee.4-1.el7.centos will be installed
--> Processing Dependency: policycoreutils-python for package: docker-ee-selinux-17.03.2.ee.4-1.el7.centos.noarch
--> Finished Dependency Resolution
Error: Package: docker-ee-selinux-17.03.2.ee.4-1.el7.centos.noarch (docker-ee-stable-17.03)
           Requires: policycoreutils-python
Error: Package: docker-ee-17.03.2.ee.4-1.el7.centos.x86_64 (docker-ee-stable-17.03)
           Requires: libseccomp.so.2()(64bit)
Error: Package: docker-ee-17.03.2.ee.4-1.el7.centos.x86_64 (docker-ee-stable-17.03)
           Requires: libltdl.so.7()(64bit)
 You could try using --skip-broken to work around the problem
 You could try running: rpm -Va --nofiles --nodigest
[root@dockerslaveone yum.repos.d]#

**Comment by liwzhi** [comment]
The same error:

yum install docker-engine
Loaded plugins: fastestmirror, langpacks
Loading mirror speeds from cached hostfile
 * epel: mirrors.tongji.edu.cn
Package docker-engine is obsoleted by docker-ce, trying to install docker-ce-18.02.0.ce-1.el7.centos.x86_64 instead
Resolving Dependencies
--> Running transaction check
---> Package docker-ce.x86_64 0:18.02.0.ce-1.el7.centos will be installed
--> Processing Dependency: container-selinux >= 2.9 for package: docker-ce-18.02.0.ce-1.el7.centos.x86_64
--> Processing Dependency: libdevmapper.so.1.02(DM_1_02_97)(64bit) for package: docker-ce-18.02.0.ce-1.el7.centos.x86_64
--> Processing Dependency: libsystemd.so.0(LIBSYSTEMD_209)(64bit) for package: docker-ce-18.02.0.ce-1.el7.centos.x86_64
--> Processing Dependency: pigz for package: docker-ce-18.02.0.ce-1.el7.centos.x86_64
--> Processing Dependency: libsystemd.so.0()(64bit) for package: docker-ce-18.02.0.ce-1.el7.centos.x86_64
--> Running transaction check
---> Package docker-ce.x86_64 0:18.02.0.ce-1.el7.centos will be installed
--> Processing Dependency: container-selinux >= 2.9 for package: docker-ce-18.02.0.ce-1.el7.centos.x86_64
--> Processing Dependency: libdevmapper.so.1.02(DM_1_02_97)(64bit) for package: docker-ce-18.02.0.ce-1.el7.centos.x86_64
--> Processing Dependency: libsystemd.so.0(LIBSYSTEMD_209)(64bit) for package: docker-ce-18.02.0.ce-1.el7.centos.x86_64
--> Processing Dependency: libsystemd.so.0()(64bit) for package: docker-ce-18.02.0.ce-1.el7.centos.x86_64
---> Package pigz.x86_64 0:2.3.4-1.el7 will be installed
--> Finished Dependency Resolution
Error: Package: docker-ce-18.02.0.ce-1.el7.centos.x86_64 (docker-ce-edge)
           Requires: libsystemd.so.0(LIBSYSTEMD_209)(64bit)
Error: Package: docker-ce-18.02.0.ce-1.el7.centos.x86_64 (docker-ce-edge)
           Requires: libdevmapper.so.1.02(DM_1_02_97)(64bit)
Error: Package: docker-ce-18.02.0.ce-1.el7.centos.x86_64 (docker-ce-edge)
           Requires: container-selinux >= 2.9
Error: Package: docker-ce-18.02.0.ce-1.el7.centos.x86_64 (docker-ce-edge)
           Requires: libsystemd.so.0()(64bit)
 You could try using --skip-broken to work around the problem
 You could try running: rpm -Va --nofiles --nodigest

**Comment by RakeshNagarajan** [comment]
Add the default repositories of centOS 7
 CentOS-Vault.repo
 CentOS-Sources.repo
 CentOS-Media.repo
 CentOS-fasttrack.repo
 CentOS-Debuginfo.repo
 CentOS-CR.repo
 CentOS-Base.repo

For me this is working :)
